packages/itwinui-css/src/slider/slider.scss

```diff
@@ -144,3 +131,118 @@ $tick-height: $iui-baseline; // 11px
     }
   }
 }
+
+@mixin iui-slider-horizontal {
+  // TODO: Vilius said this selector might not be needed and we can directly write the attributes. It works, but confirm if that is okay
+  // &.iui-slider-component-container {
+  min-height: $iui-baseline * 2;
+  // }
+
+  .iui-slider-rail {
+    width: 100%;
+    height: $track-height;
+    top: $iui-sm;
+  }
+
+  .iui-slider-thumb {
+    top: $track-height + 1; // position center of thumb in center of track
+  }
+
+  .iui-slider-track {
+    height: $track-height;
+    left: 0;
+  }
+
+  .iui-slider-ticks {
+    padding-top: $iui-m;
+
+    .iui-slider-tick {
+      line-height: $tick-height * 4;
+      margin-bottom: $tick-height * 2;
+
+      .iui-slider-tick-mark {
+        width: 1px;
+        height: $tick-height;
+      }
+
+      .iui-slider-tick-number {
+        top: $tick-height * 2;
+      }
+    }
+  }
+
+  .iui-slider-min,
+  .iui-slider-max {
+    margin-top: $iui-xxs;
+  }
+
+  .iui-slider-min {
+    margin-right: $iui-sm;
+    text-align: right;
+  }
+
+  .iui-slider-max {
+    margin-left: $iui-sm;
+  }
+}
+
+@mixin iui-slider-vertical {
+  // TODO: Vilius said this selector might not be needed and we can directly write the attributes. It works, but confirm if that is okay
+  // &.iui-slider-component-container {
+  flex-direction: column-reverse;
+  min-height: auto;
+  min-width: $iui-baseline * 2;
+  height: 100%;
```

Comment:
This doesn't seem to work. Try removing `height: 500px` from the demo and slider won't take full height. It's either problem with input container that it doesn't work properly with vertical slider or need to check other fixes. Because I removed input container wrapper from one of the vertical sliders and removed `height: 500px` then slider shrinked, then removed `height: 100%` and it expanded trough all height.

Reply:
I tried replicating what you said. After removing `height: 500px` from the `iui-slider-container`, removing the slider from the input container, and removing `height: 100%`, the vertical slider does expand completely. But that was because it was in the row flex (the div that arranges the vertical sliders horizontally).

I'm not entirely sure I understood the problem you are highlighting. But if I understood correctly, then if the slider is not in a flex (like a row flex) and not in a `display: grid` (input container) but in a container with a fixed height, then the `height: 100%` makes the vertical slider extend the entire height of the container. That way users can size the wrapping container to their requirements and the vertical/horizontal slider fits the container.

Please let me know if I misunderstood you.